commit bac23cc
Author: perlancar (on PC, Bandung) <perlancar@gmail.com>
Date:   Tue Jan 12 23:10:44 2016 +0700

    decode $end too

diff --git a/lib/Pod/Elemental/PerlMunger.pm b/lib/Pod/Elemental/PerlMunger.pm
index c1c22a9..40e3b45 100644
--- a/lib/Pod/Elemental/PerlMunger.pm
+++ b/lib/Pod/Elemental/PerlMunger.pm
@@ -169,8 +169,17 @@ around munge_perl_string => sub {
 
   s/\n\s*\z// for $new_perl, $new_pod;
 
+  my $new_end;
+  if (defined $end) {
+    $new_end = Encode::decode(
+      'utf-8',
+      $end,
+      Encode::FB_CROAK,
+    );
+  }
+
   return defined $end
-         ? "$new_perl\n\n$new_pod\n\n$end"
+         ? "$new_perl\n\n$new_pod\n\n$new_end"
          : "$new_perl\n\n__END__\n\n$new_pod\n";
 };
 
